### PR TITLE
feat(arns): add circuit breaker for calling the IO network process

### PR DESCRIPTION
**Pull request by dtfiedler**
This should help avoid slamming AO when there are intermittent issues

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -274,6 +274,29 @@ export const ARNS_RESOLVER_PRIORITY_ORDER = env
   .varOrDefault('ARNS_RESOLVER_PRIORITY_ORDER', 'on-demand,gateway')
   .split(',');
 
+export const ARNS_ON_DEMAND_CIRCUIT_BREAKER_TIMEOUT_MS = +env.varOrDefault(
+  'ARNS_ON_DEMAND_CIRCUIT_BREAKER_TIMEOUT_MS',
+  `${60 * 1000}`, // 1 minute
```
Comment (djwhitt) [inline src/config.ts:279]: 1 minute seems a bit long. How long does the CU typically take to respond?

File: src/config.ts [hunk 2]
```diff
@@ -274,6 +274,29 @@ export const ARNS_RESOLVER_PRIORITY_ORDER = env
...
+export const ARNS_ON_DEMAND_CIRCUIT_BREAKER_ERROR_THRESHOLD_PERCENTAGE =
+  +env.varOrDefault(
+    'ARNS_ON_DEMAND_CIRCUIT_BREAKER_ERROR_THRESHOLD_PERCENTAGE',
+    '50',
+  );
+
+export const ARNS_ON_DEMAND_CIRCUIT_BREAKER_ROLLING_COUNT_TIMEOUT_MS =
+  +env.varOrDefault(
+    'ARNS_ON_DEMAND_CIRCUIT_BREAKER_ROLLING_COUNT_TIMEOUT_MS',
+    `${1000 * 10}`, // 10 seconds
```
[the remark this answers is not in the document]
Comment (djwhitt) [reply src/config.ts:291]: Yeah, I agree with this.

**Review by djwhitt — APPROVED** [review]
Code looks good. Just need to nail down the timeouts.

File: src/config.ts (hunk 1 above)
Comment (dtfiedler) [reply src/config.ts:279]: for this process, should be more like ~3 seconds

**Pull request merged**